jvm/core/src/test/kotlin/com/intuit/playerui/core/player/HeadlessPlayerTest.kt

```diff
@@ -543,4 +554,33 @@ internal class HeadlessPlayerTest : PlayerTest(), ThreadUtils {
         val lastname = constantsController.getConstants(key = "lastname", namespace = "constants")
         assertEquals("doe", lastname)
     }
+
+    @TestTemplate
+    fun `beacon plugin receives logger`() = runBlockingTest {
+        var logger: LoggerType? = null;
+
+        val action = "clicked"
+        val element = "button"
+        val (id, type) = "test-id" to "test"
+
+        val asset = buildJsonObject {
+            put("id", id)
+            put("type", type)
+        }.asAsset()
+
+        beaconPlugin.hooks.cancelBeacon.tap { beaconOptions ->
+            logger = beaconOptions?.logger
+            BailResult.Bail(false)
+        }
+
+        beaconPlugin.beacon(action, element, asset) shouldBe Unit
+        while (logger == null) runBlocking { delay(100) }
+        logger?.get(LogSeverity.trace)?.invoke("Beacon Plugin trace logged") shouldBe Unit
```

Comment:
I'm not 100% sure on this, does this pass even if the functions are null? If so it's fine, otherwise i'd check the functions exist/isn't null